src/common/d3Components/tooltip/tooltip.ts

```diff
@@ -0,0 +1,58 @@
+import { D3Selection } from '../../types';
+import { cnc } from '../../helpers/classname';
+import { mouse } from 'd3';
+const styles = require('./tooltip.scss');
+
+interface Props {
+  node: D3Selection;
+  xOffset?: number;
+  yOffset?: number;
+}
+
+interface State {
+  tooltip: D3Selection;
+}
+
+export const tooltipComponent = (props: Props) => {
+  const state = createEmptyState();
+  state.tooltip = render(props);
+
+  return {
+    onShowTooltip: onShowTooltip(props, state),
+    onHideTooltip: onHideTooltip(state),
+    onUpdateTooltipPosition: onUpdateTooltipPosition(props, state),
+  };
+};
```

Comment:
This componet is a bit strange for me, could @fjcalzado or @Nasdan ellaborate in a reply how does it work?

Reply:
I think if it's a d3 component,we should suffix it, to avoid missunderstanding, should we name all d3 components something like:

tooltipD3Component